### PR TITLE
Add has_duplicates property for Index and MultiIndex

### PR DESCRIPTION
**Pull request by charlesdong1991**

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -125,6 +125,18 @@ def spark_type(self):
...
 
+    @property
+    def has_duplicates(self) -> bool:
+        """
+        If index has duplicates, return True, otherwise False.
+        """
+        idx_column = self._kdf._sdf.select(self._scol)
+        deduplicate_idx_column = idx_column.drop_duplicates()
+
+        if idx_column.count() == deduplicate_idx_column.count():
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: This runs Spark jobs twice. Can we use `F.count` and `F.countDistinct` instead?

Comment (charlesdong1991) [reply databricks/koalas/indexes.py]: ahh! Nice! thanks!!!

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -125,6 +125,34 @@ def spark_type(self):
...
+
+        >>> kdf = ks.DataFrame({'a': [1, 2, 3]}, index=[list('aac'), list('eef')])
+        >>> kdf.index.has_duplicates
+        True
+        """
+        scol = self._kdf._sdf.select(self._scol)
+        col = scol.columns[0]
+
+        count = scol.agg(F.count(col).alias('count')).collect()
+        dedup_count = scol.agg(F.countDistinct(col).alias('count')).collect()
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py]: @charlesdong1991 can we do it in one pass?

```
df.select(F.count("id") == F.countDistinct("id")).first()
```

Comment (charlesdong1991) [reply databricks/koalas/indexes.py]: Nice, this is much simpler! thanks! @HyukjinKwon

**Pull request merged**